compiler/rustc_middle/src/mir/interpret/pointer.rs

```diff
@@ -120,9 +120,10 @@ pub trait Provenance: Copy + fmt::Debug {
     where
         Self: Sized;
 
-    /// Provenance must always be able to identify the allocation this ptr points to.
+    /// Provenance must always be able to identify the allocation this ptr points to,
+    /// if `OFFSET_IS_ADDR == false`, else it's simply a best effort approach.
```

Comment:
```suggestion
    /// if `OFFSET_IS_ADDR == false`, provenance must always be able to identify
    ///  the allocation this ptr points to. Otherwise this function is never called.
```
Is that true? I guess it should be. :D